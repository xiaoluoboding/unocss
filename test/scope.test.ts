import { createGenerator } from '@unocss/core'
import presetUno from '@unocss/preset-uno'
import { expect, test } from 'vitest'

export const fixture = new Set([
  'pl-10px',
  'hover:p-4',
  '!p-5px',
  '!hover:px-10',
  'hover:!p-10',
  'flex',
  'dark:text-xl',
  'dark:hover:text-xl',
  'sm:text-red-100',
  'sm:text-red-200/10',
  'md:!hidden',
])

const uno = createGenerator({
  presets: [
    presetUno(),
  ],
})

test('scope', async() => {
  const { css } = await uno.generate(fixture, { scope: '.foo-scope' })
  expect(css).toMatchSnapshot()
})
