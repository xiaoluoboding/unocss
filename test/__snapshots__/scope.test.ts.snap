// Vitest Snapshot v1

exports[`scope 1`] = `
"/* layer: default */
.foo-scope .flex{display:flex;}
.foo-scope .\\\\!p-5px{padding:5px !important;}
.foo-scope .hover\\\\:\\\\!p-10:hover{padding:2.5rem !important;}
.foo-scope .hover\\\\:p-4:hover{padding:1rem;}
.foo-scope .\\\\!hover\\\\:px-10:hover{padding-left:2.5rem !important;padding-right:2.5rem !important;}
.foo-scope .pl-10px{padding-left:10px;}
.dark .foo-scope .dark\\\\:hover\\\\:text-xl:hover,.dark .foo-scope .dark\\\\:text-xl{font-size:1.25rem;line-height:1.75rem;}
@media (min-width: 640px){
.foo-scope .sm\\\\:text-red-100{--un-text-opacity:1;color:rgba(254,226,226,var(--un-text-opacity));}
.foo-scope .sm\\\\:text-red-200\\\\/10{color:rgba(254,202,202,0.1);}
}
@media (min-width: 768px){
.foo-scope .md\\\\:\\\\!hidden{display:none !important;}
}"
`;
